### PR TITLE
Pt

### PR DESCRIPTION
**Pull request by pabloslash**
Push LSTM.py and helper.py from pt to master.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -18,13 +18,17 @@
...
+end_char = '%'    # Every music sample <end> will be marked with %
 pad_char = '`'
-data = clean_up_data('input.txt', start_char, end_char)
+data = clean_up_data('input.txt', start_char, end_char) #Last <end> does not get replaced by %
+len_ABC_file = avg_len_music_file(data, start_char) #Avg length of music file in ABC format
+
+# import IPython; IPython.embed()
 
-vocab = get_idx(data + start_char + end_char + pad_char)
+vocab = get_idx(data + start_char + end_char + pad_char) #!!! WHY + $,%,'??
```
Comment (michaelwiest) [inline main.py]: Because they aren't present in the original corpus.

File: LSTM.py [hunk 2]
```diff
@@ -169,6 +175,8 @@ def train(self, vocab_idx, seq_len, batch_size, epochs, lr,
                 rand_slice = add_cuda_to_variable(rand_slice, self.use_gpu)
                 targets = add_cuda_to_variable(targets, self.use_gpu)
 
+                #!!!! TARGETS[0] = rand_slice[1]. Is this right???
```
Comment (michaelwiest) [inline LSTM.py]: Yes. Why are you commenting with a million exclamation points.

File: LSTM.py [hunk 3]
```diff
@@ -150,17 +152,21 @@ def train(self, vocab_idx, seq_len, batch_size, epochs, lr,
...
             '''
             while len(possible_example_indices) > self.batch_size:
+                #Get #(batch_size) random training examples to take samples from
                 example_indices = random.sample(possible_example_indices, self.batch_size)
 
                 # Get processed data.
                 # print(len(possible_slice_starts[example_indices[0]]))
                 len_old = len(possible_example_indices)
+
+                #Are we removing stuff from (possible_slice_starts)? Not a global var
```
Comment (michaelwiest) [inline LSTM.py]: It gets removed in the function as a pointer. it's not a global variable. it is passed to the function.

**Review by michaelwiest — CHANGES_REQUESTED** [review]
Remove all the music files. And unnecessary stuff.

**Comment by pabloslash** [comment]
Closing and redoing merge request.

**Pull request closed without merging**